src/components/DonutChart.js

```diff
@@ -0,0 +1,234 @@
+
+      const endAngle = this.props.chartTotal ? valueTotal / this.props.chartTotal : 1;
+
+      const pie = d3.layout.pie().sort(null).padAngle(this.props.padAngle).endAngle(endAngle * 2 * Math.PI);
+      const values = pie(dataSets);
+      const radius = (Math.min(this.props.width, this.props.height) / 2) - this.props.radiusOffset;
+      const standardArc = d3.svg.arc().outerRadius(radius - this.props.activeOffset).innerRadius(radius - this.props.arcWidth);
+      const hoverArc = d3.svg.arc().outerRadius(radius).innerRadius(radius - this.props.arcWidth);
+
+
+      return values.map((point, i) => {
+        const arc = this.props.activeIndex === i && this.props.animateOnHover ? hoverArc : standardArc;
+
+        return (
+          <g
+            key={i}
+            onClick={this.props.onClick.bind(null, i)}
+            onMouseEnter={this.props.onMouseEnter.bind(null, i)}
+            onMouseLeave={this.props.onMouseLeave}
+          >
+            <DonutPath {...this.props} arc={arc} color={this.props.colors[i]} value={point} />
+          </g>
+        );
+      });
+    }
+  },
+
+  _renderBaseArc () {
+    if (this.props.showBaseArc) {
+      const radius = (Math.min(this.props.width, this.props.height) / 2) - this.props.radiusOffset;
+      const baseArc = d3.svg.arc().outerRadius(radius - this.props.activeOffset)
+                                  .innerRadius(radius - this.props.arcWidth)
+                                  .startAngle(0)
+                                  .endAngle(2 * Math.PI);
+
+      return (
+        <g
+          key={baseArc}
+        >
+          <path d={baseArc()} fill={this.props.baseArcColor}></path>
+        </g>
+      );
+    }
+  },
+
+  _renderDataPoints () {
+    const dataPoints = this.props.dataPoints.map(dataPoint => {
+      return dataPoint.value;
+    });
+
+    const radius = (Math.min(this.props.width, this.props.height) / 2) - this.props.radiusOffset;
+
+    return dataPoints.map((dataPoint, index) => {
+      const endAngle = this.props.chartTotal ? dataPoint / this.props.chartTotal : dataPoint / 100;
+
+      const dataPointArc = d3.svg.arc()
+        .outerRadius(radius - this.props.activeOffset)
+        .innerRadius(radius - this.props.arcWidth)
+        .startAngle(0)
+        .endAngle(endAngle * 2 * 2 * Math.PI);
+
+      const position = 'translate(' + dataPointArc.centroid() + ')';
+
+      return (
+        <circle cx='0' cy='0' r={this.props.dataPointRadius} key={index} transform={position} fill={this.props.dataPointColors[index]} />
+      );
+    });
+  },
+
+  _renderLegend () {
+    if (this.props.showLegend) {
+      if (this.props.children) {
+        return (
+          <div onClick={this.props.onClick} style={styles.center}>
+            {this.props.children}
+          </div>
+        );
+      } else {
+        const labelStyle = objectAssign({
+          color: '#333',
+          fontSize: '18px',
+          fontWeight: '800'
+        }, this.props.labelStyle);
```

Comment:
Our components should be "Radium ready", so let's use the Radium method of combining styles. Refer to the other components to get an idea of how to do this.